### ISSUE TITLE
如何运用在自己的图片数据集上进行运行

### ISSUE THREAD

**Issue by gguyinging** [DeweiHu/OCT_DDPM#11]
我不知道我的理解是否正确，我想我的训练集是原图，用来进行加噪扩散和逆向扩散，我的测试集则是有噪声的一些图片，是吗？我想我的训练集是4000张.png的图片，保存在一个路径中，测试集图片如上，我不知道该如何更改？以及您程序中E：//中的tool我需要下载码？

**Comment by DeweiHu** [comment]
Yes, you should add Gaussian noise with a beta schedule to the original images and train the denoise model. Note that in the testing phase you need to try different t values to find the optimal one. Please ignore the additional system path in my code (that was the directory of the helper function file util.py).

**Comment by gguyinging** [comment]
是的，老师。我在训练和测试的数据集采取上改编了您的程序，如下。但我不知道您原本的数据集是否是图片，我的程序改完后，进入main运行还是不行，出现报错，不知道是为什么？我想问一下您是否可以给我一些指导？
![9daad98b53e6d04383763b6daecaf74](https://github.com/DeweiHu/OCT_DDPM/assets/150215001/92c18312-c3f6-4a07-b768-d5bae25f8298)
![b7ebbcd0c6c4549854408b315109f2b](https://github.com/DeweiHu/OCT_DDPM/assets/150215001/04a2a883-934d-403c-bb7b-77fbc961a773)

**Comment by DeweiHu** [comment]
You can check out this https://github.com/DeweiHu/OCT_DDPM/issues/10

**Comment by gguyinging** [comment]
好的老师，就是在OCT——dataloader中践行更改吗？

**Comment by gguyinging** [comment]
老师，你好。我遇到了在自己的图片数据集上进行运行，但是其loss value在一百左右，想问一下正常应该是多少？过大或过小可能的原因是什么？

**Comment by gguyinging** [comment]
目前程序中的loss value是不是对plot出的图片进行一个验证的损失？我可以加大验证集的数量进行一个平均？

**Comment by DeweiHu** [comment]
The plot is just a simple observation of the training process. The loss is computed by loss = noise_estimation_loss(model, x, t, e, b), the MSE between the Gaussian noise prediction model(x, t) and the real noise e.

**Comment by gguyinging** [comment]
我在想损失值几百，会不会是因为加载图片的时候没有除以255，所以过大？

**Comment by gguyinging** [comment]
我在想损失值几百，会不会是因为加载图片的时候没有除以255，所以过大？您的损失值大约在什么范围内？

**Comment by DeweiHu** [comment]
For diffusion applications, the intensity is usually normalized to [-1, 1], otherwise the scheduled small Gaussian noise is not sufficient to convert the image distribution to the pure Gaussian distribution. In this work, I normalized to [1, 3].

**Comment by gguyinging** [comment]
所以对于我的图片数据集，应该将强度归一化为【-1,1】,我将做出尝试，谢谢您的建议

**Comment by DeweiHu** [comment]
There is a more recent code for both unconditional and conditional diffusion here https://github.com/DeweiHu/Medical-Semantic-Diffusion/tree/main/src, I think it is cleaner. You can refer to it if needed.

**Comment by gguyinging** [comment]
好的，谢谢您
